shared/common-adapters/avatar.native.js

```diff
@@ -1,36 +1,77 @@
 // @flow
 
 import React, {Component} from 'react'
+import {Image, TouchableWithoutFeedback} from 'react-native'
 import type {Props} from './avatar'
-import {Box, Icon} from '../common-adapters'
+import {Box} from '../common-adapters'
+import {images} from './icon.paths.native'
+import {globalStyles} from '../styles/style-guide'
+import * as shared from './avatar.shared'
 
-export default class Avatar extends Component {
-  props: Props;
+type State = {
+  avatarLoaded: boolean,
+}
 
-  state: {
-    avatarLoaded: boolean
-  };
+export default class Avatar extends Component<void, Props, State> {
+  props: Props;
+  state: State;
 
   constructor (props: Props) {
     super(props)
     this.state = {avatarLoaded: false}
   }
 
   render () {
-    return (
-      <Box style={{justifyContent: 'flex-end', ...this.props.style}} onClick={this.props.onClick}>
-        <Icon style={{...avatarStyle(this.props.size - 2)}}
-          type='placeholder-avatar' />
+    const {size} = this.props
+    const uri = {uri: shared.createAvatarUrl(this.props)}
+    console.log('uri & state & props', uri, this.state, this.props)
+
+    const avatarImage =
+      <Box style={stylesContainer(size)}>
+        <Image
+          style={{...stylesImage(size), opacity: this.state.avatarLoaded ? 1 : 0}}
+          onLoad={() => this.setState({avatarLoaded: true})}
+          source={uri} />
+        {!this.state.avatarLoaded &&
+          <Image
+            style={stylesPlaceholderImage(size)}
+            source={images['placeholder-avatar']} />}
       </Box>
-    )
-  }
-}
 
-function avatarStyle (size: number): Object {
-  return {
-    width: size,
-    height: size,
-    borderRadius: size / 2,
-    alignSelf: 'center',
+    if (this.props.onClick) {
+      return (
+        <TouchableWithoutFeedback style={{...stylesContainer(size), ...this.props.style}} onPress={this.props.onClick}>
+          {avatarImage}
+        </TouchableWithoutFeedback>
```

Comment:
can't you use TouchableWithoutFeedback always and pass no onPress and set disabled = true?